app/controllers/CyaSupplementaryUnitController.scala

```diff
@@ -93,7 +93,10 @@ class CyaSupplementaryUnitController @Inject() (
             model.supplementaryUnit.getOrElse("")
           )
           val isValueChanged                              =
-            initialHasSuppUnitOpt.exists(_ != finalHasSuppUnit) || initialSuppUnitOpt.exists(_ != finalSuppUnit)
+            initialHasSuppUnitOpt.exists(_ != finalHasSuppUnit) || compareSupplementaryUnits(
```

Comment:
I'm think I'm finding the mix of getOrElse on some options and just using some options directly a bit confusing to follow! 

Any reason to not leave them all as options? Something like
`          val initialHasSuppUnitOpt2 =             request.session.get(initialValueOfHasSuppUnit).map(_.toBoolean)
          val initialSuppUnitOpt2 =             request.session.get(initialValueOfSuppUnit)
          val finalhasSuppUnitOpt2 = model.hasSupplementaryUnit
          val finalSuppUnitOpt2 = model.supplementaryUnit
          
                    val isValueChangedd = initialHasSuppUnitOpt2 == finalhasSuppUnitOpt2 || compareSupplementaryUnits2(initialSuppUnitOpt2, finalSuppUnitOpt2)

`

with

`  private def compareSupplementaryUnits2(initialSuppUnit: Option[String], finalSuppUnit: Option[String]): Boolean = {
    val initialSuppUnitBD = BigDecimal(initialSuppUnit.getOrElse("0"))
    val finalSuppUnitBD = BigDecimal(finalSuppUnit.getOrElse("0"))
    initialSuppUnitBD != finalSuppUnitBD
  }`
  
  Dunno, let me know what you think.

Reply:
agreed for consistency!